sdks/python/apache_beam/pvalue.py

```diff
@@ -158,6 +159,20 @@ def __ne__(self, other):
   def __hash__(self):
     return hash((self.tag, self.producer))
 
+  class PCollectionTypeConstraint(SequenceTypeConstraint):
```

Comment:
I would really prefer it if all type constraint classes were in the same module (typehints).

Reply:
Moved it back to the typehints module

Both files import from each other causing a circular import error so the current (hacky) workaround is to put the import inside of PCollectionTypeConstraint's __class_getitem__ function, which works. There are [no performance drawbacks](https://stackoverflow.com/questions/3095071/) to this approach but it's not consistent with the existing styling so I will look for a better solution.

Reply:
This PR uses the built-in Generic[T] instead of a new TypeConstraint so this is no longer relevant.